lib/new_relic/environment_report.rb

```diff
@@ -44,7 +44,11 @@ def self.registered_reporters=(logic)
     ####################################
     report_on('Gems') do
       begin
-        Bundler.rubygems.all_specs.map { |gem| "#{gem.name}(#{gem.version})" }
+        if Bundler::VERSION >= '2'
+          Bundler.rubygems.installed_specs.map { |gem| "#{gem.name}(#{gem.version})" }
+        else
+          Bundler.rubygems.all_specs.map { |gem| "#{gem.name}(#{gem.version})" }
+        end
```

Comment:
You could DRY up these ones with metaprogramming, but I completely respect the slight repetition if avoiding metaprogramming is a deliberate goal.

```suggestion
          specs_method = Bundler::VERSION >= '2' ? :installed_specs : :all_specs
          Bundler.rubygems.send(specs_method).map { |gem| "#{gem.name} (#{gem.version})" }
```

Reply:
I love this!